### PR TITLE
[snappyjs] Add types for snappyjs

### PR DESCRIPTION
**Pull request by MysteryBlokHed**
Related discussions: #68875, #68897

Adds types for the npm package snappyjs (<https://www.npmjs.com/package/snappyjs>).

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/snappyjs/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5801e3c8690e1bb73f133472e899197c7ff38434
</p>

File: types/snappyjs/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+/**
+ * @param input The byte stream to compress
+ * @returns The compressed byte stream, of the same type as the input
+ */
+export function compress<T extends ArrayBuffer | Uint8Array = ArrayBuffer | Uint8Array>(input: T): T;
+
+/**
+ * @param input The compressed input
+ * @param maxLength If provided, throws an exception if the data length in the header
```
Comment (peterblazejewicz) [inline types/snappyjs/index.d.ts:12]:
```suggestion
 * @param [maxLength] If provided, throws an exception if the data length in the header
```

**Comment by peterblazejewicz** [comment]
you aware it ships browser version? `SnappyJS`
https://github.com/zhipeng-jia/snappyjs/blob/master/dist/snappyjs.js#L35

**Comment by MysteryBlokHed** [comment]
> you aware it ships browser version? `SnappyJS`
> https://github.com/zhipeng-jia/snappyjs/blob/master/dist/snappyjs.js#L35

Oops, I missed that. I'll add a `declare namespace`.

**Review by sheetalkamat — APPROVED** [review]

**Comment by MysteryBlokHed** [comment]
Ready to merge

**Pull request merged**